File: src/Services/SalesService/SalesService.Domain/Entities/Service.cs
using NanoidDotNet;

namespace SalesService.Domain.Entities
{
    public class Service
    {
        public string Id { get; private set; } = Nanoid.Generate(size: 15);
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public decimal Price { get; set; } = 0;
        public int Tax { get; set; } = 0;
        public DateTime CreatedAt { get; private set; } = DateTime.UtcNow;
        public int? Duration { get; set; } = null;
    }
}

File: src/WebApp/smartbusiness.web/src/models/service.ts
export interface Service {
  id: string;
  companyId: string;
  name: string;
  description: string;
  category: string;
  price: number;
  tax: number;
  createdAt: string; // ISO string (DateTime)
  duration?: number | null;
}

export type NewService = Omit<Service, "id" | "companyId" | "createdAt">;

export interface ServiceContextType {
  createService: (service: NewService) => Promise<void>;
  updateService: (service: Partial<Service>) => Promise<void>;
  deleteService: (serviceId: string) => Promise<void>;
  fetchService: (serviceId: string) => Promise<Service | null>;
  fetchServices: (params: GetServiceByParamsQuery) => Promise<Service[]>;
}

export interface GetServiceByParamsQuery {
  Name?: string;
  Category?: string;
  MinPrice?: number;
  MaxPrice?: number;
  MinDuration?: number;
  MaxDuration?: number;
}